app/recipe/serializers.py

```diff
@@ -0,0 +1,52 @@
+"""
+Recipe API serializers.
+"""
+from rest_framework import serializers
+
+from core.models import Recipe, Ingredient
+
+
+class IngredientSerializer(serializers.ModelSerializer):
+    """Serializer for the ingredient model."""
+
+    class Meta:
+        model = Ingredient
+        fields = ['id', 'name']
+        read_only_fields = ['id']
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    """Serializer for the recipe model."""
+    ingredients = IngredientSerializer(many=True, required=False)
+
+    class Meta:
+        model = Recipe
+        fields = ['id', 'name', 'description', 'ingredients']
+        read_only_fields = ['id']
+
+    def _add_ingredients(self, ingredients, recipe):
+        """Get or create ingredients."""
+        for ingredient in ingredients:
+            ingredient_obj, _ = Ingredient.objects.get_or_create(**ingredient)
+            recipe.ingredients.add(ingredient_obj)
+
+    def create(self, data):
+        """Create a recipe."""
+        ingredients = data.pop('ingredients', [])
+        recipe = Recipe.objects.create(**data)
```

Comment:
Yeah this is something I was wondering about. It wasn't explicit in the exercise instructions, but you could currently add as many recipes with the same name as you want. It would probably make more sense to add some logic to prevent this, depending on the intention of the app. If the app had multiple users who were adding their own recipes, maybe they should be able to have recipes with the same name. But that's not necessarily the case here.